src/ApacheOrcDotNet.OptimizedReader/Infrastructure/HttpByteRangeProvider.cs

```diff
@@ -24,9 +23,9 @@ public int GetRange(Span<byte> buffer, long position)
             var response = _httpClient.Send(request);
 
             if (!response.Content.Headers.ContentRange.Length.HasValue)
-                throw new InvalidOperationException("Range respose must include a length.");
+                throw new InvalidOperationException("Range response must include a length.");
 
-            return DoRead(response.Content.ReadAsStream(), buffer);
+            return response.Content.ReadAsStream().Read(buffer);
```

Comment:
I think I asked you to make this change, but I was wrong.  I don't this is safe.  I think `Read` calls on `NetworkStream` may return less data than you requested.  When you're using the results of these calls you're assuming the buffer is fully filled which may not be the case with this implementation.

I think we need to go back to the `DoRead` or `ReadAll` functionality.  But also do we need to return a value here, if we're guaranteeing that the entire buffer will be filled?

Reply:
Since we always provide the buffers with the expected length, I've removed the return values and updated the interface to be `FillBuffer` methods.

If we had gone back to the `DoRead` methods, we would no longer be able to guarantee that the buffer was filled and some extra slicing could be required.